corehq/apps/users/role_utils.py

```diff
@@ -33,22 +18,23 @@ def archive_custom_roles_for_domain(domain):
 
 
 def unarchive_roles_for_domain(domain):
-    all_roles = UserRole.by_domain(domain, include_archived=True)
+    all_roles = SQLUserRole.objects.filter(domain=domain, is_archived=True)
     for role in all_roles:
-        if role.is_archived:
-            role.is_archived = False
-            role.save()
+        role.is_archived = False
+        role.save()
 
 
 def reset_initial_roles_for_domain(domain):
-    initial_roles = [x for x in UserRole.by_domain(domain) if x.name in UserRolePresets.INITIAL_ROLES]
+    initial_roles = [
+        role for role in SQLUserRole.objects.get_by_domain(domain)
+        if role.name in UserRolePresets.INITIAL_ROLES
+    ]
     for role in initial_roles:
-        role.permissions = UserRolePresets.get_permissions(role.name)
-        role.save()
+        role.set_permissions(UserRolePresets.get_permissions(role.name).to_list())
+        role._migration_do_sync()  # sync role to couch
```

Comment:
how come this needs to do the sync, and the previous util does `.save()`

Reply:
this could also call `save` on the SQL role but it's not strictly necessary to re-save the role. We do need to sync to couch however since `set_permissions` will not update the couch role.